### ISSUE TITLE
Error running demo.m

### ISSUE THREAD

**Issue by mrkulk** [arron2003/rgbd2full3d#1]
I get the following error running the code. -- 

Preprocessing RGBD data..
Computing Normal..
Computing Room Rotation..
Parsing with NYU segmentation..
Extracting SIFT..
Fitting planes..
Computing PB..
Merging Level 1..Error using pca (line 324)
The estimated covariance matrix is not positive semi-definite. Consider using
'complete' instead of 'pairwise' option.

Error in princomp (line 26)
    varargout{1} = pca(varargin{1},'Algorithm','eig','Economy',fEconomy);

Error in fit_plane_pca (line 10)
  coeff = princomp(points3d);

Error in NYUparse>get_features_3d (line 307)
    surfaceNormals(ii,:) = fit_plane_pca(samplePoints{ii});

Error in NYUparse (line 60)
    features3d = get_features_3d(boundaryInfo, featureData, points3d, normals,
    regionInteriors);

Error in demo (line 23)
load(['cache/' opt.name '.mat']); data=NYUparse(data, opt);

**Comment by arron2003** [comment]
Sorry for the very late reply. 

I believe they stem from the changed behavior of new versions of matlab and it will be trivial to fix them (I was running 2009b).
For PCA warnings, you can change from princomp(...) to "pca(...,'Algorithm','svd','Economy',false)"
For ismember warnings, you can change to "ismember(..., 'legacy')"

Unfortunately I do not own a newer version of matlab to fully test it. I shall fix these issue once I have access to newer version of matlab.

**Comment by mrkulk** [comment]
Yes I tested that the same day and everything works fine! Thanks...

**Comment by jiunyen-ching** [comment]
Hi @arron2003 @mrkulk, do you still have the files downloaded via setup.m? I tried downloading them but the links are now broken.

**Comment by arron2003** [comment]
You can download it at https://dhoiem.cs.illinois.edu/publications/iccv13-cleaned_3D.zip

**Comment by jiunyen-ching** [comment]
Thanks for the new link @arron2003! 

I tried running `pipeline_preprocess.m` by placing `nyu_depth_v2_labeled.mat` and `support_labels.mat` downloaded from 

> http://horatio.cs.nyu.edu/mit/silberman/indoor_seg_sup/

as such

![image](https://user-images.githubusercontent.com/25114361/111968044-b8813300-8b33-11eb-86ba-ae10a16a4547.png)

but I ran into an error saying that `Subscripted assignment dimension mismatch`.

Full stack trace:
`>> pipeline_preprocess`
`The size of scenes = 464`
`Subscripted assignment dimension mismatch.`

`Error in processRoom (line 8)`
  `d(opt.mask(1):opt.mask(3),opt.mask(2):opt.mask(4)) = data.depths;`

`Error in pipeline_preprocess (line 79)`
  `data = processRoom(data);`

I tried changing dimensions of `d` in `processRoom.m` from `d = zeros(opt.orig_H, opt.orig_W);` to `d = zeros(opt.H, opt.W);` to match the dimensions of `depths` variable in `t` struct but I still get the same error. Is there any other changes that I could try?
